cowprotocol/accounting/rewards/main_rewards_dashboard_query_2510345.sql

```diff
@@ -1,29 +1,33 @@
 with
 
 block_range as (
-    select * from "query_3333356(blockchain='ethereum',start_time='{{start_time}}',end_time='{{end_time}}')"
+    select * from "query_3333356(blockchain='{{blockchain}}',start_time='{{start_time}}',end_time='{{end_time}}')"
 ),
 
 solver_slippage as (
     select
         solver_address as solver,
         slippage_wei * 1.0 / pow(10, 18) as slippage
-    from "query_4070065(blockchain='ethereum',start_time='{{start_time}}',end_time='{{end_time}}',slippage_table_name='slippage_per_solver')"
+    from "query_4070065(blockchain='{{blockchain}}',start_time='{{start_time}}',end_time='{{end_time}}',slippage_table_name='slippage_per_solver')"
 ),
 
 named_results as (
-    select * from "query_1541516(blockchain='ethereum',end_time='{{end_time}}',vouch_cte_name='named_results')"
+    select * from "query_1541516(blockchain='{{blockchain}}',end_time='{{end_time}}',vouch_cte_name='named_results')"
 ),
 
 -- BEGIN SOLVER REWARDS
 batch_rewards as (
-    select  --noqa: ST06
-        from_hex(rbr.solver) as winning_solver,
-        -- Unpacking the needed data
-        cast(cast(rbr.data.fee as varchar) as int256) as fee,  -- noqa: RF01
-        cast(cast(rbr.data.execution_cost as varchar) as int256) as execution_cost,  -- noqa: RF01
-        cast(cast(rbr.data.capped_payment as varchar) as int256) as capped_payment  -- noqa: RF01
-    from cowswap.raw_batch_rewards as rbr
+    select
+        rbr.solver as winning_solver,
```

Comment:
Why don't we need to use the `from_hex()` and `cast()` methods anymore?

Reply:
For the solver, the reason we were using from_hex() was that the `solver` was stored as a string (and not as varbinary) in the cowswap.raw_batch_rewards table.

As for the different casts for fees, executions costs etc, i think they were just unnecessary, and an overkill. It looks like we wanted to get to an int256 but the queries also work with the decimal(38,0) type that is what we use here anyways.

Definitely this is not very well-thought through, so we could also make the raw order/batch tables return int256 instead.